src/scripts/compile.ts

```diff
@@ -77,12 +80,25 @@ export const registerHandlers = (types: Handler[]): { [type: string]: HandlerFun
 	}, {});
 };
 
+export const getFrontmatter = (content: string): { [key: string]: string } => {
```

Comment:
Could we combine this with `fromMarkdown` to avoid parsing the markdown twice?

Simply return both the content and the meta as a tuple.

```
return [
	result.children.find((node: any) => isVNode(node))
	result.children.find((node: any) => node.type === 'yaml')
];
```

```
const [ content, meta ] = await fromMarkdown(rawContent, registerHandlers(handlers));
```

Or if there will only ever be the two children, find may be unnecessary.

Reply:
I originally tried a combined approach, but parsed frontmatter nodes were lost during the `remark2rehype` processing phase. I will dig deeper to see if we can get around this.

Reply:
After additional investigation, I'm not sure we can easily combine these two methods. From https://github.com/syntax-tree/mdast-util-to-hast#notes:

> yaml and toml nodes are ignored

It seems like two separate pipelines will be required for us to handle frontmatter while still supporting rehype.